common/djangoapps/course_category/models.py

```diff
@@ -83,6 +83,25 @@ def save_reindex_course_category(sender, instance, pk_set, action, **kwargs):
         instance.pre_clear_course_keys.update(str(x.id) for x in instance.courses.all())
 
     if action in ['post_add', 'post_remove']:
-        courses_set.update(str(x.id) for x in instance.courses.all())
+        courses_set.update(str(c.id) for c in instance.courses.all())
         courses_set.update(getattr(instance, 'pre_clear_course_keys', set()))
         task_reindex_courses.delay(instance.id, list(courses_set))
+
+
+@receiver(post_init, sender=CourseCategory)
+def remember_previous_name(sender, instance, **kwargs):
+    '''
+    Saves category name to check if it was changed.
+    '''
+    instance.previous_name = instance.name
```

Comment:
I prefer to not do it in the signal. It is not obvious.
But for this fix - let's leave it as is for now.

Reply:
Got it!